src/ext/java/org/opentripplanner/ext/vectortiles/layers/vehicleparkings/VehicleParkingGroupsLayerBuilder.java

```diff
@@ -0,0 +1,66 @@
+package org.opentripplanner.ext.vectortiles.layers.vehicleparkings;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.Point;
+import org.opentripplanner.ext.vectortiles.LayerBuilder;
+import org.opentripplanner.ext.vectortiles.PropertyMapper;
+import org.opentripplanner.ext.vectortiles.VectorTilesResource;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.vehicle_parking.VehicleParkingService;
+import org.opentripplanner.transit.service.TransitService;
+import org.opentripplanner.util.geometry.GeometryUtils;
+
+public class VehicleParkingGroupsLayerBuilder extends LayerBuilder<VehicleParkingAndGroup> {
+
+  static Map<VehicleParkingGroupsLayerBuilder.MapperType, Function<Graph, PropertyMapper<VehicleParkingAndGroup>>> mappers = Map.of(
+    VehicleParkingGroupsLayerBuilder.MapperType.Digitransit,
+    g -> DigitransitVehicleParkingGroupPropertyMapper.create()
+  );
+  private final Graph graph;
+
+  public VehicleParkingGroupsLayerBuilder(
+    Graph graph,
+    TransitService transitService,
+    VectorTilesResource.LayerParameters layerParameters
+  ) {
+    super(
+      layerParameters.name(),
+      mappers
+        .get(VehicleParkingGroupsLayerBuilder.MapperType.valueOf(layerParameters.mapper()))
+        .apply(graph)
+    );
+    this.graph = graph;
+  }
+
+  @Override
+  protected List<Geometry> getGeometries(Envelope query) {
+    VehicleParkingService service = graph.getVehicleParkingService();
+    if (service == null) {
+      return List.of();
+    }
+    return service
+      .getVehicleParkingGroups()
+      .entrySet()
+      .stream()
+      .map(vehicleParkingGroupEntry -> {
+        var group = vehicleParkingGroupEntry.getKey();
+        Coordinate coordinate = new Coordinate(group.x(), group.y());
+        Point point = GeometryUtils.getGeometryFactory().createPoint(coordinate);
+        var parking = vehicleParkingGroupEntry.getValue();
+        var parkingAndGroup = new VehicleParkingAndGroup(group, parking);
+        point.setUserData(parkingAndGroup);
+        return point;
+      })
+      .collect(Collectors.toList());
```

Comment:
```suggestion
      .toList();
```

Reply:
Seems like there is a problem with the generics if you use toList(). Sorry for the confusion, you can revert the change.

Reply:
I added a cast to Geometry type to fix the compile issue. I don't know which is cleaner but I don't mind leaving it like it is.

Reply:
Thanks!